modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/features/DocumentationProviderFeatures.java

```diff
@@ -0,0 +1,172 @@
+package org.openapitools.codegen.languages.features;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * The DocumentationProvider Features support to additional properties to select the
+ * documentation provider and the annotation library to use during code generation.
+ *
+ * @author cachescrubber, 2022-01-08
+ * @since 5.4.0
+ */
+public interface DocumentationProviderFeatures {
+
+  String DOCUMENTATION_PROVIDER = "documentationProvider";
+
+  String ANNOTATION_LIBRARY = "annotationLibrary";
+
+  /**
+   * Define the default documentation Provider for CliOpts processing.
+   * Override in subclasses to customize.
+   * @return the default documentation provider
+   */
+  default DocumentationProvider defaultDocumentationProvider() {
+    return DocumentationProvider.NONE;
```

Comment:
The concern I would have is the configuration looks misleading to other java generators that have not adopted the `DocumentationProvidersFeatures` interface - e.g. `java` with `resttemplate` set will generate Swagger1 annotations, but if I tried to set the configuration options it would say it's not supported. Maybe that's fine, just kind of a weird use case until all generators are updated

Reply:
I think that would be good, so that it doesn't look incorrect for Java generators that do not yet support the new option.

Overall I'm in favor of this change to allow more standard adoption of annotation providers - as long as it's done in a way that is backwards compatible.  But I'm also not a Java technical committee member. @wing328 any thoughts?

Reply:
The last commit allows sub classes to opt-out (which is the default) of the DocumentationProvider Feature.

Reply:
Agreed with @welshm and thanks for adding the change for the sub classes to opt-out.